pkg/volume/util/operationexecutor/operation_generator_test.go

```diff
@@ -0,0 +1,250 @@
+	"k8s.io/kubernetes/pkg/volume/csi"
+	"k8s.io/kubernetes/pkg/volume/gcepd"
+	volumetesting "k8s.io/kubernetes/pkg/volume/testing"
+	"os"
+	"testing"
+)
+
+// this method just tests the volume plugin name that's used in CompleteFunc, the same plugin is also used inside the
+// generated func so there is no need to test the plugin name that's used inside generated function
+func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
+	type testcase struct {
+		name                  string
+		isCsiMigrationEnabled bool
+		pluginName            string
+		csiDriverName         string
+		csiMigrationFeature   featuregate.Feature
+		pvSpec                v1.PersistentVolumeSpec
+		probVolumePlugins     []volume.VolumePlugin
+	}
+
+	testcases := []testcase{
+		{
+			name:                  "gce pd plugin: csi migration disabled",
+			isCsiMigrationEnabled: false,
+			pluginName:            plugins.GCEPDInTreePluginName,
+			pvSpec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{},
+				}},
+			probVolumePlugins: gcepd.ProbeVolumePlugins(),
+		},
+		{
+			name:                  "gce pd plugin: csi migration enabled",
+			isCsiMigrationEnabled: true,
+			pluginName:            plugins.GCEPDInTreePluginName,
+			csiDriverName:         plugins.GCEPDDriverName,
+			csiMigrationFeature:   features.CSIMigrationGCE,
+			pvSpec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{},
+				}},
+			probVolumePlugins: gcepd.ProbeVolumePlugins(),
+		},
+		{
+			name:                  "aws ebs plugin: csi migration disabled",
+			isCsiMigrationEnabled: false,
+			pluginName:            plugins.AWSEBSInTreePluginName,
+			pvSpec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{},
+				}},
+			probVolumePlugins: awsebs.ProbeVolumePlugins(),
+		},
+		{
+			name:                  "aws ebs plugin: csi migration enabled",
+			isCsiMigrationEnabled: true,
+			pluginName:            plugins.AWSEBSInTreePluginName,
+			csiDriverName:         plugins.AWSEBSDriverName,
+			csiMigrationFeature:   features.CSIMigrationAWS,
+			pvSpec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{},
+				}},
+			probVolumePlugins: awsebs.ProbeVolumePlugins(),
+		},
+	}
+
+	for _, tc := range testcases {
+		expectedPluginName := tc.pluginName
+		if tc.isCsiMigrationEnabled {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, true)()
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, tc.csiMigrationFeature, true)()
+			expectedPluginName = fmt.Sprintf("%s:%s", csi.CSIPluginName, tc.csiDriverName)
+		}
```

Comment:
`if !tc.isCsiMigrationEnabled` we should explicitly turn off the feature gates in preparation for whenever the gates are on by default.

Reply:
fixed, thanks!